src/ripple/shamap/SHAMap.h

```diff
@@ -182,10 +182,9 @@ class SHAMap
     
     bool getRootNode (Serializer & s, SHANodeFormat format) const;
     std::vector<uint256> getNeededHashes (int max, SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (uint256 const& hash, Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
+    SHAMapAddNode addRootNode (boost::optional<uint256> const& hash,
```

Comment:
I don't think it makes much difference here.None of the callers pass in a `boost::optional`, so it's going to have to make one either way. Change it?